crates/matrix-sdk/src/widget_api/widget.rs

```diff
@@ -0,0 +1,31 @@
+//! The description of a widget.
+
+use tokio::sync::mpsc::{Receiver, Sender};
+
+/// Describes a widget.
+#[derive(Debug)]
+pub struct Widget {
+    /// Information about the widget.
+    pub info: Info,
+    /// Communication channels with a widget.
+    pub comm: Comm,
+}
+
+/// Information about a widget.
+#[derive(Debug)]
+pub struct Info {
+    /// Widget's unique identifier.
+    pub id: String,
+    /// Whether or not the widget should be initialized on load message (`ContentLoad` message),
+    /// or upon creation/attaching of the widget to the SDK's state machine that drives the API.
+    pub init_on_load: bool,
```

Comment:
This seems more like a config thing than an "info" thing.. Anyways I'd be willing to merge this module as-is and change it in follow-up PRs, when it's clearer how everything fits together.

Reply:
Yeah, indeed.